app/controllers/records_controller.rb

```diff
@@ -44,13 +46,14 @@ def download_mpl
     if BSON::ObjectId.legal?(params[:format])
       bundle = Bundle.find(BSON::ObjectId.from_string(params[:format]))
 
-      unless File.exist?(bundle.mpl_path)
-        MplDownloadCreateJob.perform_now(bundle.id.to_s)
+      if bundle.mpl_status != :ready
+        flash[:info] = 'This bundle is currently preparing for download.'
+        redirect_to :back
```

Comment:
If we just called mpl_prepare instead of mpl_status here then that would be taken care of. The main reason I did not is because mpl_prepare is already called from the mpl_download page.